ruma-client-api/src/r0/session/sso_login_with_provider.rs

```diff
@@ -0,0 +1,73 @@
+//! [GET /_matrix/client/r0/login/sso/redirect/{idp_id}](https://github.com/matrix-org/matrix-doc/blob/master/proposals/2858-Multiple-SSO-Identity-Providers.md)
+//!
+//! This uses the unstable prefix in [MSC2858](https://github.com/matrix-org/matrix-doc/pull/2858).
+#![cfg(feature = "unstable-pre-spec")]
+#![cfg_attr(docsrs, doc(cfg(feature = "unstable-pre-spec")))]
+
+use ruma_api::ruma_api;
+
+ruma_api! {
+    metadata: {
+        description: "Get the SSO login identity provider url.",
+        method: GET,
+        name: "sso_login_with_provider",
+        path: "/_matrix/client/unstable/org.matrix.msc2858/login/sso/redirect/:idp_id",
+        rate_limited: false,
+        authentication: None,
+
+    }
+
+    request: {
+        /// The ID of the provider to use for SSO login.
+        #[ruma_api(path)]
+        pub idp_id: &'a str,
```

Comment:
Is this one of the providers above, e.g. `org.matrix.twitter`?

Reply:
No, it's an opaque string defined by the homeserver. This corresponds to the `id` field of `IdentityProvider`.

For reference, Synapse uses `oidc-[brand]`.

Reply:
Okay, then it should definitely just be a string